examples/weather/unified_recipe/train.py

```diff
@@ -0,0 +1,358 @@
+
+                    # Switch to eval mode
+                    model.eval()
+
+                    # Validation loop
+                    loss_epoch = 0.0
+                    num_examples = 0
+                    for i, data in enumerate(val_datapipe):
+
+                        # Get predicted and unpredicted variables
+                        predicted_variables = data[0]["predicted"]
+                        unpredicted_variables = data[0]["unpredicted"]
+
+                        # Normalize variables
+                        predicted_variables = normalize_variables(predicted_variables, predicted_batch_norm)
+                        unpredicted_variables = normalize_variables(unpredicted_variables, unpredicted_batch_norm)
+
+                        # Perform validation step and compute loss
+                        loss, net_predicted_variables, predicted_variables = eval_forward(model, predicted_variables, unpredicted_variables, cfg.training.nr_input_steps)
+                        loss_epoch += loss.detach().cpu().numpy()
+                        num_examples += predicted_variables.shape[0]
+
+                        # Plot validation on first batch
+                        if i == 0:
+                            net_predicted_variables = net_predicted_variables.cpu().numpy()
+                            predicted_variables = predicted_variables.cpu().numpy()
+                            for chan in range(net_predicted_variables.shape[2]):
+                                plt.close("all")
+                                fig, ax = plt.subplots(
+                                    3, net_predicted_variables.shape[1], figsize=(15, net_predicted_variables.shape[0] * 5)
+                                )
+                                for t in range(net_predicted_variables.shape[1]):
+                                    ax[0, t].set_title("Network prediction, Step {}".format(t))
+                                    ax[1, t].set_title("Ground truth, Step {}".format(t))
+                                    ax[2, t].set_title("Difference, Step {}".format(t))
+                                    ax[0, t].imshow(net_predicted_variables[0, t, chan])
+                                    ax[1, t].imshow(predicted_variables[0, t, chan])
+                                    ax[2, t].imshow(net_predicted_variables[0, t, chan] - predicted_variables[0, t, chan])
+                
+                                fig.savefig(f"forcast_validation_channel{chan}_epoch{epoch}.png")
+
+                    # Log validation loss
+                    log.log_epoch({"Validation error": loss_epoch / num_examples})
+
+                    # Switch back to train mode
+                    model.train()
+
+            # Sync after each epoch
+            if dist.world_size > 1:
+                torch.distributed.barrier()
+
+            # Save checkpoint
+            if (epoch % 5 == 0 or epoch == 1) and dist.rank == 0:
```

Comment:
Can this be configurable?

Reply:
Ill worry about this in a future PR for this.